src/test/java/tests/TurboMilestoneTest.java

```diff
@@ -34,6 +32,45 @@ public void turboMilestoneTest() {
     }
 
     @Test
+    public void getDueDateComparator_combinationOfDifferentMilestones_correctMilestoneOrdering() {
```

Comment:
I'm not sure myself. Let's discuss on slack and decide which one we should go for.